Makefile

```diff
@@ -164,9 +172,31 @@ docker-multi-stage:
 	@echo ">> building docker image 'thanos' with Dockerfile.multi-stage"
 	@docker build -f Dockerfile.multi-stage -t "thanos" --build-arg BASE_DOCKER_SHA=$(BASE_DOCKER_SHA) .
 
-.PHONY: docker-push
+GET_SHA = $(shell echo '$1'_SHA | tr '[:lower:]' '[:upper:]')
+# docker-build builds docker images with multiple architectures.
+.PHONY: docker-build $(BUILD_DOCKER_ARCHS)
+docker-build: $(BUILD_DOCKER_ARCHS)
+$(BUILD_DOCKER_ARCHS): docker-build-%:
+	@docker build -t "$(DOCKER_IMAGE_REPO)-linux-$*:$(DOCKER_IMAGE_TAG)" \
```

Comment:
I think the problem is the same: here we are copying the same `thanos` binary into both images. We need to have the OS/ARCH as parameters https://github.com/prometheus/prometheus/blob/main/Dockerfile#L8 and use them when copying the binary. So, it should be something like:
```
make crossbuild # To build all binaries
```

And then `Dockerfile` needs to be like that example (:

Reply:
In fact Prometheus only build Linux OS:
https://github.com/prometheus/prometheus/blob/main/Makefile.common#L249-L256

Reply:
Now updated.